lasagne/layers/dnn.py

```diff
@@ -10,16 +9,30 @@
 from .normalization import BatchNormLayer
 from ..utils import as_tuple
 
-if not theano.sandbox.cuda.cuda_enabled:
-    raise ImportError(
+dnn_avail = False
+try:
+    from theano import gpuarray
+except ImportError:
+    from theano.sandbox import gpuarray
+if gpuarray.pygpu_activated and gpuarray.dnn.dnn_present():
+    dnn = gpuarray.dnn
+    dnn_avail = True
+else:
+    try:
+        from theano.sandbox import cuda
+
+        enabled = cuda.cuda_enabled
+        if enabled and cuda.dnn.dnn_available():
+            dnn = cuda.dnn
+            dnn_avail = True
+    except ImportError:
+        raise ImportError(
             "requires GPU support -- see http://lasagne.readthedocs.org/en/"
             "latest/user/installation.html#gpu-support")  # pragma: no cover
-elif not dnn.dnn_available():
-    raise ImportError(
-            "cuDNN not available: %s\nSee http://lasagne.readthedocs.org/en/"
-            "latest/user/installation.html#cudnn" %
-            dnn.dnn_available.msg)  # pragma: no cover
-
+if not dnn_avail:
+        raise ImportError(
+            "cuDNN not available:See http://lasagne.readthedocs.org\
+            /en/latest/user/installation.html#cudnn")  # pragma: no cover
```

Comment:
You're still missing the message here. In line with the suggestion above, I would suggest:
```python
# check if Theano's new GPU backend is available and in use
try:
    from theano import gpuarray as gpu
except ImportError:
    from theano.sandbox import gpuarray as gpu
gpu_enabled = gpu.pygpu_activated
dnn_enabled = gpu.dnn.dnn_present
# if not, try to fall back to Theano's old GPU backend
if not gpu_enabled:
    try:
        from theano.sandbox import cuda as gpu
    except ImportError:
        gpu_enabled = False  # [I know it's False already but it is easier to follow]
    else:
        gpu_enabled = gpu.cuda_enabled
        dnn_enabled = gpu.dnn.dnn_available
# if either of the backends is available, use it, otherwise bail out
if gpu_enabled:
    if dnn_enabled():
        dnn = gpu.dnn
    else:
        raise ImportError("cuDNN not ..." % dnn_enabled.msg)
else:
    raise ImportError("requires GPU ...")
```